Add TabButton component with TypeScript props and function handling

Tarachand90 opened the pull request:
- Defined TabButtonProps interface for type safety.
- Created TabButton component that accepts children and an onButtonClick function.
- Passed functions as props to TabButton in App.tsx to handle button clicks dynamically.
- Integrated TabButton into the App component for interactive tab selection.

Tarachand90 committed:
Add TabButton component with TypeScript props and function handling

- Defined TabButtonProps interface for type safety.
- Created TabButton component that accepts children and an onButtonClick function.
- Passed functions as props to TabButton in App.tsx to handle button clicks dynamically.
- Integrated TabButton into the App component for interactive tab selection.

DebaKaran commented on src/components/TabButtonProps.ts:
```diff
@@ -0,0 +1,8 @@
+import { ReactNode } from 'react';
+
+interface TabButtonProps {
```
Use named export instead of default one as you have defined it as an interface.

Tarachand90 replied:
Done, Similarly changed the CoreConceptsType(interface) from default to named export

DebaKaran commented on src/components/TabButton.tsx:
```diff
@@ -0,0 +1,11 @@
+const TabButton = ({ children, onButtonClick }: TabButtonProps) => {
+  return (
+    <li>
+      <button onClick={onButtonClick}>{children}</button>
```
Wrap button text i.e children in <span> for better styling

DebaKaran commented on src/App.tsx:
```diff
@@ -1,14 +1,36 @@
           <h2>Time to get started!</h2>
           <CoreConceptList />
+          <h2>Examples</h2>
+          <menu>
```
Moved this code to a different Component "TabMenu" and refractor it by using map method

DebaKaran commented on src/App.tsx:
```diff
@@ -1,14 +1,36 @@
+              State
+            </TabButton>
+          </menu>
+          Dynamic Content
```
Remove this statement

DebaKaran commented on src/App.tsx:
```diff
@@ -1,14 +1,36 @@
+import TabButton from './components/TabButton';
 
 function App() {
+  const onClickHandler = (selectedButton) => {
```
Here selectedButton is explicitly defined as any, change it to correct type.

Tarachand90 committed:
Refactor TabMenu component

- Moved tab button rendering to a separate TabMenu component
- Passed activeTab prop to TabMenu for state control
-  Updated TabButton to highlight the active tab dynamically
- Improved separation of concerns and code reusability

DebaKaran reviewed (commented):
Make the changes

DebaKaran reviewed (approved):
Looks fine! Approved it.
